File: TwoSum/TwoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let numMap = new Map(); // Create a new Map

    for (let i = 0; i < nums.length; i++) {
        let complement = target - nums[i]; // Find the required number

        if (numMap.has(complement)) { // Check if the complement exists in the Map
            return [numMap.get(complement), i]; // Return indices
        }

        numMap.set(nums[i], i); // Store the current number with its index
    }

    return []; // If no solution is found (shouldn't happen with valid input)
};

// Test case
//console.log(twoSum([2, 5, 5, 11], 10)); // Output: [1, 2]

File: TwoSum/TwoSum.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        prevMap = {}  # Dictionary to store {value: index}

        for i, n in enumerate(nums):
            diff = target - n  # Compute the complement

            if diff in prevMap:  # Check if complement exists in the dictionary
                return [prevMap[diff], i]  # Return indices

            prevMap[n] = i  # Store the current number with its index

        return []  # Return empty list if no solution is found

# Create an instance of Solution
sol = Solution()

# Test the function with an example input
nums = [2, 5, 5, 11]
target = 10
result = sol.twoSum(nums, target)

# Print the output
print(result)  # Expected Output: [1, 2]
